### ISSUE TITLE
onnx_tf.backend.prepare(model) in Tensorflow to ONNX tutorial error: "InvalidArgumentError: Dimensions must be equal"

### ISSUE THREAD

**Issue by ividal** [onnx/tutorials#41]
Python version: 3.5.2
onnx==1.2.1
onnx-tf==1.1.2
tensorflow-gpu==1.8.0
Using tutorial as of [this commit](https://github.com/onnx/tutorials/commit/70d57df667e494d8fb07130f334cb58f4e3f04ac).

Following the instructions in he tutorial, I've used [this script](https://github.com/onnx/tutorials/blob/master/tutorials/assets/tf-train-mnist.py) to train. Worked smoothly.
I froze the model using:

```
python3 /path/to/site-packages/tensorflow/python/tools/freeze_graph.py \
    --input_graph=/home/ividal/dev/onnx/tutorials/tutorials/graph.proto \
    --input_checkpoint=/home/ividal/dev/onnx/tutorials/tutorials/ckpt/model.ckpt \
    --output_graph=/tmp/frozen_graph.pb \
    --output_node_names=fc2/add \
    --input_binary=True
``` 
This produced the expected `/tmp/frozen_graph.pb` .
The export code in the tutorial provides the expected `mnist.onnx` file.

`model = onnx.load('mnist.onnx')` works, but:

`tf_rep = prepare(model)` yields:

```
---------------------------------------------------------------------------
InvalidArgumentError                      Traceback (most recent call last)
~/.venvs/onnx/lib/python3.5/site-packages/tensorflow/python/framework/ops.py in _create_c_op(graph, node_def, inputs, control_inputs)
   1566   try:
-> 1567     c_op = c_api.TF_FinishOperation(op_desc)
   1568   except errors.InvalidArgumentError as e:

InvalidArgumentError: Dimensions must be equal, but are 16 and 64 for 'Add_1' (op: 'Add') with input shapes: [?,64,?,16], [1,1,1,64].
```

From the error message, I gather the expected channels might be switched (?). However, I did not modify the tutorial code, so it shouldn't be that. Any ideas...?

Thanks!

**Comment by ividal** [comment]
Just in case I repeated everything with Tensorflow 1.5.0, since it's the last version explicitly mentioned in the documentation, but the error is exactly the same.

**Comment by Revo-Future** [comment]
i meet the same problem with u, does anyone has some solution??

**Comment by ividal** [comment]
I have a feeling it's [this problem](https://github.com/tensorflow/tensorflow/issues/13146). NCHW vs NHWC at the different steps: training vs freezing vs exporting vs loading in onnx. Just don't know exactly where or how to fix it.

**Comment by knandanan** [comment]
@ividal Did you find a solution to this issue yet? Please let me know.

**Comment by mukul74** [comment]
Did anyone found the solution to the issue ?

**Comment by amsalien** [comment]
Did anyone found the solution to the issue ?

**Comment by ividal** [comment]
@knandanan No, sorry, opted to keep onnx and TF separate for this (I stick to TF and a .pb if deployment had to be with TF, e.g. an android device).
